app/src/main/java/net/squanchy/settings/view/SettingsHeaderLayout.java

```diff
@@ -0,0 +1,112 @@
+
+public class SettingsHeaderLayout extends LinearLayout {
+
+    @Nullable
+    private ImageLoader imageLoader;
+
+    private ImageView userPhotoView;
+    private TextView userNameView;
+
+    public SettingsHeaderLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public SettingsHeaderLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+        this(context, attrs, defStyleAttr, 0);
+    }
+
+    public SettingsHeaderLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
+        super(context, attrs, defStyleAttr, defStyleRes);
+
+        if (!isInEditMode()) {
+            imageLoader = ImageLoaderInjector.obtain(unwrapToActivityContext(context))
+                    .imageLoader();
+        }
+
+        super.setOrientation(VERTICAL);
+    }
+
+    @Override
+    public void setOrientation(int orientation) {
+        throw new UnsupportedOperationException("SettingsHeaderLayout doesn't support changing orientation");
+    }
+
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+
+        userPhotoView = (ImageView) findViewById(R.id.user_photo);
+        userNameView = (TextView) findViewById(R.id.user_name);
+    }
+
+    public void updateWith(Optional<FirebaseUser> user) {
+        if (user.isPresent() && !user.get().isAnonymous()) {
+            updateWithAuthenticatedUser(user.get());
+        } else {
+            updateWithNoOrAnonymousUser();
+        }
+    }
+
+    private void updateWithAuthenticatedUser(FirebaseUser firebaseUser) {
+        Optional<UserInfo> googleUserInfo = googleUserInfoFrom(firebaseUser);
+        if (googleUserInfo.isPresent()) {
+            UserInfo userInfo = googleUserInfo.get();
+            updateUserPhotoFrom(userInfo);
+            userNameView.setText(userInfo.getDisplayName());
+        }
+    }
+
+    private Optional<UserInfo> googleUserInfoFrom(FirebaseUser firebaseUser) {
+        List<? extends UserInfo> providerData = firebaseUser.getProviderData();
+        List<? extends UserInfo> googleData = Lists.filter(providerData, data -> "google.com".equalsIgnoreCase(data.getProviderId()));
+        if (googleData.isEmpty()) {
+            return Optional.absent();
+        }
+        return Optional.of(googleData.get(0));
+    }
+
+    private void updateUserPhotoFrom(UserInfo userInfo) {
+        if (imageLoader == null) {
+            return;
+        }
+
+        Optional<String> photoUrl = photoUrlFor(userInfo);
+        if (photoUrl.isPresent()) {
+            imageLoader.load(photoUrl.get())
+                    .into(userPhotoView);
+        }
+    }
+
+    private Optional<String> photoUrlFor(UserInfo userInfo) {
+        return Optional.fromNullable(userInfo.getPhotoUrl())
+                .map(Uri::toString);
+    }
+
+    private void updateWithNoOrAnonymousUser() {
+        userPhotoView.setImageDrawable(null);
```

Comment:
Can we instead add a default icon? I think it could be easier for the user to just see the usual anonymous icon, instead of an empty view